pkg/api/pss.go

```diff
@@ -86,6 +82,20 @@ func (s *server) pssPostHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func (s *server) pssWsHandler(w http.ResponseWriter, r *http.Request) {
+	upgrader := websocket.Upgrader{
+		ReadBufferSize:  swarm.ChunkSize,
+		WriteBufferSize: swarm.ChunkSize,
+		CheckOrigin: func(r *http.Request) bool {
+			o := r.Header.Get("Origin")
+			if o != "" {
```

Comment:
this will always return true when origin is set. you'd probably want to get rid of this statement and just have the following one. I also think that `containsOrigin` might need to compare whether the origin header value has the compared item's value as a substring, since origin header can [contain port also](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin), it means that even if you define `localhost` as an allowed origin, it will not compare since the request came in with `localhost:1234` value on the actual header.

Reply:
if i understand right first point is as above i think?

second point - i think you are right - i have taken the condition directly from [here](https://github.com/ethersphere/bee/blob/2b491c9f6e7279d1c8928ad03f68ce1f6ee05b41/pkg/api/router.go#L169), the CORS is port specific so just `localhost` would imply `localhost:80`. 

so i could add `80` if port ommitted, or could add a check to include the port number in the config, as we do with api-addr et al?

Reply:
i'm not sure at all that the port incoming on the `Origin` header is `80` at all. If anything I think it should reflect the incoming request port and my guess is that this would be a high port. It would be cool if you could test it with the websocket on the browser side, and just print the header when you're in the upgrader (add `fmt.Println(r.Header))`) to see which port we get there. If I remember correctly when we were both testing it the port was a high port